### DISCUSSION TITLE
RFC: Add parallel execution of npm commands on generated projects

### DISCUSSION THREAD

**Discussion by bterone** [nimblehq/react-templates#140] [RFCs]
## Issue

Currently a newly generated project for running linting and fixes all execute synchronously ⬇️ 

```js
{
	...,
	"scripts": {
		...,
    	"codebase:lint": "npm run lint && npm run stylelint",
	    "codebase:fix": "npm run lint:fix && npm run stylelint:fix",
	}
}
```

However, there are libraries to run these commands in parallel and offer useful glob patterns like [npm-run-all](https://www.npmjs.com/package/npm-run-all) or [concurrently](https://www.npmjs.com/package/concurrently) 🤩 

## Solution

It's a good idea to integrate one of these libraries into our own template 🤗 

Currently, on the project that we're working on, we use [**concurrently**](https://www.npmjs.com/package/concurrently)  for linting, fixes, and tests

```js
{
	"lint": "concurrently yarn:lint:scripts yarn:lint:styles",
	"fix": "concurrently yarn:lint:scripts:fix yarn:lint:styles:fix",
	"ci": "concurrently yarn:lint yarn:test:unit yarn:test:ui",
}
```

However, a newly generated [Remix](https://remix.run/) project will use [**npm-run-all**](https://www.npmjs.com/package/npm-run-all) and make use of different functionality like `run-p` for parallel running and `run-s` for sequential commands but that project hasn't seen any active development for **4 years now**.

```js
{
	"build": "run-s build:*",
	"dev": "run-p dev:*",
	"validate": "run-p \"test -- --run\" lint typecheck test:e2e:run"
}
```

It offers a much nicer developer experience to use either of these libraries and simplifies commands by a lot by cutting out the `npm run x && npm run y` chains that can appear when a project gets larger or has a lot of commands ✨ 

## Who Benefits?

Developers ❤️ and future template users 😳 

## What's next?

- [x] Gather feedback
- [x] Integrate one of the concurrent libraries to the template
- [x] Update commands in `package.json` to be simplified, E.g. `concurrently npm:lint:*` or `run-p lint:*`

**Comment by olivierobert** [comment]
As you pointed out, the main issue with `npm-run-all` seems to be that it is no longer actively maintained. Just added some screenshots for the others to have more insights:

![image](https://user-images.githubusercontent.com/696529/211759854-e5cce36b-bb95-4af8-ab22-7db0ae34c11f.png)

![image](https://user-images.githubusercontent.com/696529/211759980-f61168d7-6c80-427e-9355-8943293307d9.png)

**Comment by bterone** [comment]
Went with the [concurrently](https://www.npmjs.com/package/concurrently) package and merged in #149

**Comment by tyrro** [comment]
I've checked the two checkboxes for you @bterone in the `What's next` section 😉 I guess you can also close this RFC as well 🙌🏽

**Reply by bterone** [reply]
Thanks, I will now close this RFC 🙌
